queries/gdshader/highlights.scm

```diff
@@ -0,0 +1,123 @@
+[
+  "render_mode"
+  "shader_type"
+  "group_uniforms"
+  "global"
+  "instance"
+  "const"
+  "varying"
+  "uniform"
+  "struct"
+  "for"
+  "in"
+  "out"
+  "inout"
+  "while"
+  "do"
+  "switch"
+  "case"
+  "default"
+  "if"
+  "else"
+  "elif"
+  "continue"
+  "break"
+  "return"
+  "include"
+  "#"
+  (precision_qualifier)
+  (interpolation_qualifier)
+] @keyword
+
+(string) @string
+
+[
+  "="
+  "+="
+  "-="
+  "!"
+  "~"
+  "+"
+  "-"
+  "*"
+  "/"
+  "%"
+  "||"
+  "&&"
+  "|"
+  "^"
+  "&"
+  "=="
+  "!="
+  ">"
+  ">="
+  "<="
+  "<"
+  "<<"
+  ">>"
+  "++"
+  "--"
+] @operator
+
+[
+  (boolean)
+  (integer)
+  (float)
+] @number
+
+[
+  "."
+  ","
+  ";"
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.delimiter
+
+(builtin_type) @type
+
+(ident_type) @type.definition
+
+[
+  (shader_type)
+  (render_mode)
+  (hint_name)
+] @property
```

Comment:
At the top level of each GDShader file, you declare a 'shader_type,' which can be one of five builtin tokens. You also do the same for 'render_mode', which can be a list of ~20 builtin tokens. Hint names are flags that let the engine itself know that it can set them in a particular editor hint. 

uniform vec4 color: source_color;

Here, 'source_color' is a hint_name. Maybe property isn't the best word for them, but in terms of highlighting I think it makes sense that these 3 categories are all the same color.

Reply:
`@attribute` could work.